activerecord/lib/active_record/migration.rb

```diff
@@ -678,17 +680,26 @@ def disable_ddl_transaction!
       end
 
       def check_pending_migrations # :nodoc:
-        prev_db_config = Base.connection_db_config
-
-        db_configs_in_current_env.each do |db_config|
-          Base.establish_connection(db_config)
-          check_pending!
+        if pending_migrations.flatten.any?
+          raise ActiveRecord::PendingMigrationError.new(pending_migrations: pending_migrations.flatten)
```

Comment:
This loops over all the connections a second time when raising an error, right?  Is there any downside to that?

Reply:
It's not looping over the connections again, it's passing the list of pending migrations to the error message so we no longer have to interact with the connection at all once we hit the error.

Reply:
Oh I see what you're saying now. I've fixed this in https://github.com/rails/rails/pull/46421. Thanks!